src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -136,6 +136,19 @@ class IRSuite extends SparkSuite {
     assertEvalsTo(If(True(), NA(TInt32()), I32(7)), null)
   }
 
+  @Test def testIfWithDifferentRequiredness() {
+    val t = TStruct("foo" -> TArray(TInt32Required, required = true))
+    val value = Row(FastIndexedSeq(1, 2, 3))
+    assertEvalsTo(
```

Comment:
assertEvalsTo checks the non-optimized evaluation too, right? Otherwise this gets optimized away.

Reply:
well, I guess it doesn't actually check the non-optimized evaluation (for compiled functions, which is where the region offset problem would come up). But as long as the PruneDeadFields step comes after the simplify it wouldn't get optimized away anyways, right?

Reply:
correct. I can change the condition if you want, to something that evaluates a random number?

Reply:
Sure, that works. I will sometimes do something like `ArrayLen(GetField(In(0, t), "foo")).ceq(3)` or something that's pretty obviously true but statically non-optimizable, but I think this is probably also fine until physical types go in and we change the simplify rules back again?

Reply:
oh, awesome. I just changed it to an `In` bool

Reply:
that makes way more sense.